Sources_v3/Workers/Background/NewChannelQueryUpdater_Tests.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient
+import XCTest
+
+class NewChannelQueryUpdater_Tests: StressTestCase {
+    typealias ExtraData = DefaultDataTypes
+    
+    private var env: TestEnvironment!
+    
+    var database: DatabaseContainerMock!
+    var webSocketClient: WebSocketClientMock!
+    var apiClient: APIClientMock!
+    
+    var newChannelQueryUpdater: NewChannelQueryUpdater<ExtraData>!
+    
+    override func setUp() {
+        super.setUp()
+        env = TestEnvironment()
+        
+        database = try! DatabaseContainerMock(kind: .inMemory)
+        webSocketClient = WebSocketClientMock()
+        apiClient = APIClientMock()
+        
+        newChannelQueryUpdater = NewChannelQueryUpdater(database: database,
+                                                        webSocketClient: webSocketClient,
+                                                        apiClient: apiClient,
+                                                        env: env.environment)
+    }
+    
+    override func tearDown() {
+        apiClient.cleanUp()
+        
+        weak var weak_env = env
+        weak var weak_database = database
+        weak var weak_webSocketClient = webSocketClient
+        weak var weak_apiClient = apiClient
+        weak var weak_newChannelQueryUpdater = newChannelQueryUpdater
+        
+        env = nil
+        database = nil
+        webSocketClient = nil
+        apiClient = nil
+        newChannelQueryUpdater = nil
+        
+        // We need to assert asynchronously, because there can be some callbacks happening
+        // on the background queue, that keeps the worker alive, until they have finished.
+        AssertAsync {
+            Assert.willBeNil(weak_env)
+            Assert.willBeNil(weak_database)
+            Assert.willBeNil(weak_webSocketClient)
+            Assert.willBeNil(weak_apiClient)
+            Assert.willBeNil(weak_newChannelQueryUpdater)
+        }
+        
+        super.tearDown()
+    }
+    
+    func test_update_called_for_each_query() throws {
+        let filter1: Filter = .contains(.unique, String.unique)
+        let filter2: Filter = .notEqual(.unique, to: 1)
+        
+        try database.createChannelListQuery(filter: filter1)
+        try database.createChannelListQuery(filter: filter2)
+                
+        try database.createChannel()
+        
+        AssertAsync.willBeEqual(env!.channelQueryUpdater?.update_calls_counter, 2)
+    }
+    
+    func test_filter_is_Modified() throws {
+        let cid: ChannelId = .unique
+        let filter: Filter = .notEqual(.unique, to: 1)
+        
+        try database.createChannelListQuery(filter: filter)
+        try database.createChannel(cid: cid)
+        
+        let expectedFilter: Filter = .and([filter, .equal("cid", to: cid)])
+        
+        AssertAsync.willBeEqual(env!.channelQueryUpdater?.update_query?.filter.description, expectedFilter.description)
```

Comment:
Can we also assert that even though the filter was modified, the hash stays the same?

Reply:
It's due to special behavior of our `technical` enum case.
It returns filter hash from original `Filter`, but description taken from the modified filter that is inside this enum.

Reply:
But the hash should be also the same, no? I think we should verify that:
`env!.channelQueryUpdater?.update_query?.filter.hash == expectedFilter.hash`

Reply:
We keep the original` filterHash` of DB query so later we can fetch it to link to our new channel.

Reply:
And our technical enum returns this original `filterHash`, not the hash of underlying filter.